apps/minifront/src/components/shared/non-native-fee-warning.tsx

```diff
@@ -38,6 +40,9 @@ const hasTokenBalance = ({
       getAddressIndex.optional()(asset)?.account === account,
   );
 
+  // Set the staking token status in the state
+  setStakingToken(hasStakingToken);
```

Comment:
issue: the `hasTokenBalance` function is not meant to modify the state. Apart from that, it is hard for me to follow the flow of data from `gasPrices` and `stakingToken` that you added to the state.

suggestion: instead of adding new values to the send and swap Zustand states, move it to the shared state `src/state/shared.ts`. It exists for this very case, when the data from some query is shared between pages. 

You can copy the `assets` zQuery slice and modify it a little to request the `gasPrices`. This way you will eliminate the need to request and set the state in a purely informative component (non-native-fee-warning). As for the `hasStakingToken`, it could just be a hook based on the shared `gasPrices` and `balancesResponses` states

Reply:
great suggestion, I overlooked the shared state